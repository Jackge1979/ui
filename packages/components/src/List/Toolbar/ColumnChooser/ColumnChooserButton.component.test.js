import React from 'react';
import { shallow } from 'enzyme';
import ColumnChooserButton from './ColumnChooserButton.component';

describe('ColumnChooserButton', () => {
	it('should return the component (shallow mode)', () => {
		// given
		const id = 'myIdColumnChooserId';
		const columns = [
			{
				label: 'col1',
				order: 1,
				locked: false,
				hidden: false,
			},
			{
				label: 'col2',
				order: 1,
				locked: false,
				hidden: false,
			},
		];
		const handlerColumnChooser = jest.fn();
		// when
		const wrapper = shallow(
			<ColumnChooserButton id={id} columns={columns} handlerColumnChooser={handlerColumnChooser} />,
		);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
