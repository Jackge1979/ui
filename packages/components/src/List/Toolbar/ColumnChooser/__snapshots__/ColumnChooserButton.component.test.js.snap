// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChooserButton should return the component (shallow mode) 1`] = `
<Nav
  bsClass="nav"
  justified={false}
  pullLeft={false}
  pullRight={false}
  stacked={false}
>
  <Translate(ActionButton)
    aria-label="Column chooser button"
    data-feature="open-column-chooser-overlay-action"
    icon="talend-folder"
    id="myIdColumnChooserId-column-chooser-button"
    label=""
    link={true}
    overlayComponent={
      <ColumnChooserContent
        columns={
          Array [
            Object {
              "hidden": false,
              "label": "col1",
              "locked": false,
              "order": 1,
            },
            Object {
              "hidden": false,
              "label": "col2",
              "locked": false,
              "order": 1,
            },
          ]
        }
        id="myIdColumnChooserId-column-chooser-content"
        submitColumnChooser={[MockFunction]}
        t={[Function]}
      />
    }
    overlayId="myIdColumnChooserId-column-chooser-overlay"
    overlayPlacement="bottom"
  />
</Nav>
`;
